File: app/src/main/java/com/hidesign/ported/intro/SplashScreen.java
package com.hidesign.ported.intro;

import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.hidesign.ported.MainActivity;
import com.hidesign.ported.R;

import java.util.Objects;
import java.util.Timer;
import java.util.TimerTask;

import timber.log.Timber;

public class SplashScreen extends AppCompatActivity {

    FirebaseUser currentUser;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);

        FirebaseAuth mAuth = FirebaseAuth.getInstance();
        currentUser = mAuth.getCurrentUser();

        ShowMain();
    }

    public void ShowMain() {
        Timber.e("ShowMain: ");
        TimerTask task = new TimerTask() {
            @Override
            public void run() {
                if (currentUser != null) {
                    GoToMain();
                }
                else {
                    GoToSignIn();
                }
            }
        };
        new Timer().schedule(task, 2000);
    }

    private void GoToMain(){
        Intent intent = new Intent(this, MainActivity.class);
        startActivityForResult(intent, RESULT_OK);
        Objects.requireNonNull(this).finish();
    }

    private void GoToSignIn(){
        Intent intent = new Intent(this, LoginRegister.class);
        startActivityForResult(intent, RESULT_OK);
        Objects.requireNonNull(this).finish();
    }
}

File: app/src/main/java/com/hidesign/ported/intro/SplashScreen.kt
package com.hidesign.ported.intro

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.hidesign.ported.MainActivity
import com.hidesign.ported.R
import java.util.*

class SplashScreen : AppCompatActivity() {
    var currentUser: FirebaseUser? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        val mAuth = FirebaseAuth.getInstance()
        currentUser = mAuth.currentUser
        showMain()
    }

    private fun showMain() {
        val task: TimerTask = object : TimerTask() {
            override fun run() {
                if (currentUser != null) {
                    goToMain()
                } else {
                    goToSignIn()
                }
            }
        }
        Timer().schedule(task, 2000)
    }

    private fun goToMain() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }

    private fun goToSignIn() {
        val intent = Intent(this, LoginRegister::class.java)
        startActivity(intent)
        finish()
    }
}